wrapper/src/test/java/integration/refactored/container/tests/AutoscalingTests.java

```diff
@@ -0,0 +1,240 @@
+import integration.refactored.container.ConnectionStringHelper;
+import integration.refactored.container.TestDriverProvider;
+import integration.refactored.container.TestEnvironment;
+import integration.refactored.container.condition.EnableOnDatabaseEngineDeployment;
+import integration.refactored.container.condition.EnableOnNumOfInstances;
+import integration.refactored.container.condition.EnableOnTestFeature;
+import integration.refactored.container.condition.MakeSureFirstInstanceWriter;
+import integration.util.AuroraTestUtility;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import software.amazon.jdbc.ConnectionProviderManager;
+import software.amazon.jdbc.HikariPoolConfigurator;
+import software.amazon.jdbc.HikariPooledConnectionProvider;
+import software.amazon.jdbc.PropertyDefinition;
+import software.amazon.jdbc.hostlistprovider.AuroraHostListProvider;
+import software.amazon.jdbc.plugin.failover.FailoverSuccessSQLException;
+import software.amazon.jdbc.plugin.readwritesplitting.ReadWriteSplittingPlugin;
+
+@TestMethodOrder(MethodOrderer.MethodName.class)
+@ExtendWith(TestDriverProvider.class)
+@EnableOnTestFeature(TestEnvironmentFeatures.RUN_AUTOSCALING_TESTS_ONLY)
+@EnableOnDatabaseEngineDeployment({DatabaseEngineDeployment.AURORA})
+@EnableOnNumOfInstances(min = 5)
+@MakeSureFirstInstanceWriter
+public class AutoscalingTests {
+  protected static final AuroraTestUtility auroraUtil =
+      new AuroraTestUtility(TestEnvironment.getCurrent().getInfo().getAuroraRegion());
+  private static final Logger LOGGER = Logger.getLogger(AutoscalingTests.class.getName());
+
+  protected static Properties getProxiedProps() {
+    final Properties props = getProps();
+    AuroraHostListProvider.CLUSTER_INSTANCE_HOST_PATTERN.set(props,
+        "?." + TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo()
+            .getInstanceEndpointSuffix());
+    return props;
+  }
+
+  protected static Properties getDefaultPropsNoPlugins() {
+    final Properties props = ConnectionStringHelper.getDefaultProperties();
+    DriverHelper.setSocketTimeout(props, 10, TimeUnit.SECONDS);
+    DriverHelper.setConnectTimeout(props, 10, TimeUnit.SECONDS);
+    return props;
+  }
+
+  protected static Properties getProps() {
+    final Properties props = getDefaultPropsNoPlugins();
+    PropertyDefinition.PLUGINS.set(props, "auroraHostList,readWriteSplitting");
+    return props;
+  }
+
+  protected static Properties getPropsWithFailover() {
+    final Properties props = getDefaultPropsNoPlugins();
+    PropertyDefinition.PLUGINS.set(props, "readWriteSplitting,failover,efm");
```

Comment:
Do you need to set monitoring-socketTimeouts here?

Reply:
Don't think so because we are setting the regular socket timeout in getDefaultPropsNoPlugins